pkg/fission-cli/cmd/spec/list.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright 2019 The Fission Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/pkg/errors"
+
+	fv1 "github.com/fission/fission/pkg/apis/fission.io/v1"
+	"github.com/fission/fission/pkg/fission-cli/cliwrapper/cli"
+	"github.com/fission/fission/pkg/fission-cli/cmd"
+	flagkey "github.com/fission/fission/pkg/fission-cli/flag/key"
+	"github.com/fission/fission/pkg/fission-cli/util"
+)
+
+// ListSubCommand struct
+type ListSubCommand struct {
+	cmd.CommandActioner
+}
+
+// List lists resources in the spec.
+func List(input cli.Input) error {
+	return (&ListSubCommand{}).do(input)
+}
+
+func (opts *ListSubCommand) do(input cli.Input) error {
+	return opts.run(input)
+}
+
+func (opts *ListSubCommand) run(input cli.Input) error {
+	// get specdir
+	specDir := util.GetSpecDir(input)
+
+	// read everything from spec
+	fr, err := ReadSpecs(specDir)
+	if err != nil {
+		return errors.Wrap(err, "error reading specs")
+	}
+
+	fns, err := opts.Client().V1().Function().List("")
```

Comment:
Let's move the resource list & print code to it's own function instead of putting all of them in run function.
For example, 
```
func showAppliedFunctions(deployID string) {
...
}
```
In this way, we are able to write tests for different resources.